build/git.go

```diff
@@ -3,23 +3,32 @@ package build
 import (
 	"context"
 	"os"
-	"os/exec"
 	"path/filepath"
 	"strings"
 
-	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/docker/buildx/util/gitutil"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
 const DockerfileLabel = "com.docker.image.source.entrypoint"
 
-func addGitProvenance(ctx context.Context, contextPath string, dockerfilePath string) (map[string]string, error) {
-	v := os.Getenv("BUILDX_GIT_LABELS")
-	if (v != "1" && v != "full") || contextPath == "" {
+type gitProvenance struct {
+	Source         string
+	Revision       string
+	DockerfilePath string
+}
+
+func getGitProvenance(ctx context.Context, mode string, contextPath string, dockerfilePath string) (*gitProvenance, error) {
+	switch mode {
+	case "min":
+		mode = "1"
+	case "max":
+		mode = "full"
+	}
+	if (mode != "1" && mode != "full") || contextPath == "" {
```

Comment:
@tonistiigi @jedevc Atm I just translate provenance mode to the one we currently have for labels. WDYT?

Reply:
I think provenance vcs attributes should be set only for `mode=max`.

Reply:
> Atm I just translate provenance mode to the one we currently have for labels. WDYT?

We should do the reverse IMO. Let's use the `min` and `max` here, and then translate to the "legacy" labels mode.

Sidenote: we should also use `strconv.ParseBool` instead of just a check for `1`.

> I think provenance vcs attributes should be set only for `mode=max`.

Agreed I think VCS only for `mode=max` is a good call and makes sense with the rest of how `min` and `max` are setup. `min` shouldn't "leak" information, `max` should provide everything that it is possible to provide.